src/commands/aem/rde/delete.js

```diff
@@ -63,13 +68,16 @@ class DeleteCommand extends BaseCommand {
         const typeInfo = types.length === 1 ? types[0] : 'artifact';
         const serviceInfo =
           services.length === 1 ? `the ${services[0]} of ` : '';
-        cli.log(
-          `Could not delete ${typeInfo} "${args.id}". It is not present on ${serviceInfo}this environment.`
-        );
+        throw new deploymentErrorCodes.DELETE_NOT_FOUND({
+          messageValues: [typeInfo, args.id, serviceInfo],
+        });
       }
     } catch (err) {
       spinner.stop();
-      cli.log(err);
+      if (err instanceof AioError) {
+        throw err;
+      }
+      throw new internalCodes.INTERNAL_DELETE_ERROR({ messageValues: err });
```

Comment:
Would it make sense to factor out the differentiation between `AioError` and internal error? 

```suggestion
handleError(err, internalCodes.INTERNAL_HISTORY_ERROR);
```

or even

```suggestion
internalCodes.INTERNAL_HISTORY_ERROR.handle(err);
```

Reply:
Good point - managed at https://github.com/adobe/aio-cli-plugin-aem-rde/pull/53/commits/63f5067cc54d1aa5d8aeb659d7ea3cc01fe6d3b5